File: terminal-management/server/src/main/java/com/terminal/management/dto/RegisterRequest.java
package com.terminal.management.dto;

import com.fasterxml.jackson.databind.JsonNode;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RegisterRequest {

    private String deviceId;
    private String name;
    private String hostname;
    private String osVersion;
    private String cpuModel;
    private Long memoryTotal;
    private Long diskTotal;
    private String ipAddress;
    private JsonNode hardwareInfo;
    private JsonNode systemInfo;
}

File: terminal-management/client/TerminalAgent/Models/RegisterRequest.cs
namespace TerminalAgent.Models;

public class RegisterRequest
{
    public string DeviceId { get; set; } = string.Empty;
    public string HostName { get; set; } = string.Empty;
    public HardwareInfo Hardware { get; set; } = new();
    public SystemInfo System { get; set; } = new();
    public DateTime RegisteredAt { get; set; } = DateTime.UtcNow;
}
